elro/auth.py

```diff
@@ -48,6 +51,9 @@ class ElroConnectsSession:
     def __init__(self) -> None:
         """Initialize."""
         self._session_cache = None
+        self._domain = self._get_domain()
```

Comment:
We should not do IO here. Initialize to `None` here. Suggestion is to move it to `async_login` and call `_get_domain` using loop.run_in_executor().
See https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor

Reply:
Changed initialization to None, removed IO